core/domain/state_domain.py

```diff
@@ -95,13 +95,41 @@ def from_dict(cls, answer_group_dict):
         Returns:
             AnswerGroup. The corresponding AnswerGroup domain object.
         """
+        if not isinstance(answer_group_dict['rule_specs'], list):
```

Comment:
Add a unit test for this new check*

Reply:
We already test it here `def test_cannot_update_answer_groups_with_non_list_rule_specs(self)`.